### PR TITLE
Continuous deployment

### LINKED ISSUE #53: Automatic Docker builds

### LINKED ISSUE #54: Push dockerimages to registry
We also need to chose a package registry, probably either Github Package Registry or Dockerhub

### LINKED ISSUE #55: Server should automatically pull and run docker images

### PR DESCRIPTION
**Pull request by jlndk**
~Does not work. Dont merge~
This PR adds Continous Deployment to our pipeline.
It is triggered every time a commit is added to master (which also includes merges), and works by first building the docker image and afterwards push it to the docker registry.
Afterwards the action SSH into the server, pulls git, and runs our deployment script.

We could improve this pipeline by using the docker socket over ssh (using the -h argument) instead of running commands directly on the server to improve reliability and eliminate the need for git/ a local version of the source code on the server. I however think that we should merge this and improve afterwards, so that we can actually have CD.

Even though we have some degree of docker caching in this PR, I won't say it is enough to solve #90. Currently we only push the "prod" stage of our docker image to the registry, thus only being able to download that for caching. If we want full caching we should also push/pull the "build" stage seperatly.

Solves #53.
Solves #54.
Solves #55.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+on:
+    push:
+        branches:
+            - master
+
+jobs:
+    deploy:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v2
```
Comment (Jakobis) [inline .github/workflows/deploy.yml:12]: As someone not terribly well versed in this: What is the checkout@v2?

**Review by Jakobis — COMMENTED** [review]
Looks fine, and I think I get most of it. Obviously can't say if you're missing something.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (jlndk) [reply .github/workflows/deploy.yml:12]: This is simply version 2 of a github action that is used to "checkout" the code (pull it). The cool thing about external github actions is that they correspond to a repository. That means that every time you're in doubt what a action does, you can simply look at the repository corresponding to the action (example: actions/checkout -> https://github.com/actions/checkout).

**Review by Duckapple — APPROVED** [review]
I read it, and I get most of it. I'll say go.

**Review by Hjaltesorgenfrei — APPROVED** [review]
Very good

**Pull request merged**